src/server.cpp

```diff
@@ -387,9 +387,7 @@ static UA_StatusCode valueSourceRead(
     assert(nodeContext != nullptr && value != nullptr);
     auto& callback = static_cast<detail::NodeContext*>(nodeContext)->dataSource.read;
     if (callback) {
-        auto result = detail::tryInvoke(
-            callback, asWrapper<DataValue>(*value), asRange(range), includeSourceTimestamp
-        );
+        auto result = detail::tryInvoke(callback, NodeId(*nodeId), asWrapper<DataValue>(*value), asRange(range), includeSourceTimestamp);
```

Comment:
Using the `NodeId` constructor will make a copy. Casting the native type to its wrapper type with `asWrapper<NodeId>(*nodeId)` avoids the copy.
Please add `nodeId != nullptr` to the assert statement above, so we don't accidentally dereference a `nullptr`.

Reply:
Fixed